### ISSUE TITLE
Get cluster assignment (id) of each vector after IVF

### ISSUE THREAD

**Issue by frances1005** [facebookresearch/faiss#2486]
# Summary

After I have trained my index with IVF+HNSW, I'd like to get the centroid id/cluster that each vector was assigned during training in order to do further analyses on each vector's cluster assignment.
How can I do that from the trained ivf index?

Running on:
- [ x] CPU
- [ ] GPU

Interface: 
- [ ] C++
- [ x] Python

**Comment by jrcavani** [comment]
These should help you.
https://gist.github.com/mdouze/b0a65aba70a64b00d6425b0e268a4c80
https://github.com/facebookresearch/faiss/blob/main/contrib/inspect_tools.py

If vector is already added, and you would like to inspect what's inside of the inverted list, refer to those above.

If you have a new vector, do
```
D, I = index.quantizer.search(vec, 1)
```

The nearest neighbor `I` from the coarse quantizer is the assigned centroid/cluster ID. `D` also shows you the distance/similiarity.

If your index has `PreTransforms`, you would need to apply that to your vec first before doing
```
D, I = faiss.extract_index_ivf(index).quantizer.search(vec, 1)
```

**Comment by frances1005** [comment]
Perfect, thank you so much @jrcavani!

**Comment by rahchuenmonroe** [comment]
@jrcavani are both methods supposed to return the same values? I have tried with both and compared the answers and I found that in 1.6% of cases (vectors), the centroid returned is different depending on the method I have used. 
I have a set of vectors that have been already added but I only need the centroids for a subset of vectors only. 
Therefore I have stacked my vectors and did the the search `D, I = index.quantizer.search(vectors, 1)` like the way you mentioned. However to verify, I also inspected the invlist and am now seeing that difference.
